packages/core/src/client.ts

```diff
@@ -245,7 +235,63 @@ export class Client extends AsyncEventEmitter<ManagerShardEventsMap> {
 			},
 		});
 
-		return promise;
+		try {
+			for await (const payload of on(this, GatewayDispatchEvents.GuildMembersChunk, { signal: controller.signal })) {
+				const [{ data }] = payload as MappedEvents[GatewayDispatchEvents.GuildMembersChunk];
+
+				if (data.nonce !== nonce) continue;
+
+				clearTimeout(timer);
+
+				yield {
+					members: data.members,
+					nonce,
+					notFound: data.not_found ?? null,
+					presences: data.presences ?? null,
+					chunkIndex: data.chunk_index,
+					chunkCount: data.chunk_count,
+				};
+
+				if (data.chunk_index >= data.chunk_count - 1) {
+					break;
+				} else {
+					timer = createTimer();
+				}
+			}
+		} catch (error) {
+			if (error instanceof Error && error.name === 'AbortError') {
+				throw new Error('Request timed out');
+			}
+
+			throw error;
+		}
```

Comment:
We should add a `finally {}` to clear the timer in case the user `throw`s or `return`s inside the loop.